### ISSUE TITLE
BatchNorm layer not working in keras3

### ISSUE THREAD

**Issue by martin-gorner** [keras-team/keras#19072]
Repro Colab using Keras3:
https://colab.research.google.com/drive/1ZBE4jA_Wl0XkFqJrAvKRsXGDPkLAAoGw#scrollTo=-z2e9X78wAOu
In the notebook, MNIS training converges when not using BatchNorm, but does not converge withBatchNorm.

Repro Colab using Keras2: 
https://colab.research.google.com/drive/12tv_Nfqf9G62Kbx0n7GJEIGDjnQT5AsX#scrollTo=8Hzctgy0HTmv
In this notebook, both BatchNorm and non-BatchNorm runs are successsful.

I have also tested this using the Jax backend, with identical results.

Possibly related: https://github.com/keras-team/keras/issues/18594

**Comment by fchollet** [comment]
This is now fixed. Root cause was an XLA numerical instability.
